traffic_monitor/tmclient/tmclient.go

```diff
@@ -198,6 +199,10 @@ func (c *TMClient) ConfigDoc() (handler.OpsConfig, error) {
 func (c *TMClient) getBytes(path string) ([]byte, error) {
 	url := c.url + path
 	httpClient := http.Client{Timeout: c.timeout}
+	// add the http.Transport object if specified.
+	if c.Transport != nil {
```

Comment:
Nitpick: the httpClient `Transport` is nil by default here, so the `if` check isn't necessary

Reply:
@rob05c I added this so that I can inject an http.Transport into the httpClient.  I only want to inject an http.Transport when I've created and added one to the TMClient which is what's done when using a Proxy.

Reply:
@rob05c it's needed so as not to eliminate the Default Transport.  It overrides the Default Transport with the one created for using a proxy.  If I use this without the nil check, the default Transport is eliminated and the TM integrations tests fail:

httpClient := http.Client{Timeout: c.timeout, Transport: c.Transport}

Reply:
Ah, because `TMClient.Transport` a `*http.Transport` and `http.Client` is a `http.RoundTripper`, there's an implicit cast happening, and it's creating a non-nil interface pointing to a nil pointer. Which is where the weirdness and test failure is coming from.

If `TMClient.Transport` was a `RoundTripper` it'd avoid the cast, and work as-expected. But interfaces are really pointers, and pointers-to-pointers where the outer interface pointer is non-nil with an object that's a nil pointer get weird.

That's fine, I'd just leave it. Sorry, I wouldn't have suggested it if I'd noticed the types were different. Non-nil interfaces with potentially nil objects are one of Go's pain points, not worth dealing with.